refactor-challenge/clean-code_Aditya-and-Ayush/src/models/Order.model.ts

```
@@ -0,0 +1,34 @@
+import { Order } from "../classes/Order";
+import { errorMessages } from "../common/constants/errors";
+
+export class OrderModel {
+  private static orders: Map<string, Order> = new Map();
+
+  public static set(order: Order): void {
+    if (order instanceof Order && !this.orders.has(order.id)) {
+      this.orders.set(order.id, order);
+    } else {
+      throw new Error(errorMessages.invalidInstance("Order"));
```

we try to set an already existing ID; then it will throw  same error of invalidInstance of Order, which should be the case.